third_party/tlx/dialect/triton_tlx.cc

```diff
@@ -227,6 +227,34 @@ void init_triton_tlx_ir(py::module &&m) {
              Value pred = self.create<arith::ConstantIntOp>(1, 1);
              self.create<ttng::TMEMStoreOp>(dst, src, pred);
            })
+      .def("create_tcgen5_dot",
+           [](TritonOpBuilder &self, mlir::Value &a, mlir::Value &b,
+              mlir::Value &d, std::optional<Value> mBarrier) -> mlir::Value {
+             // try to find the TMEMAllocOp that created d
+             ttng::TMEMAllocOp tmemAllocOp;
+             auto value = d;
+             while (true) {
+               if ((tmemAllocOp = value.getDefiningOp<ttng::TMEMAllocOp>())) {
+                 break;
+               }
+               auto subviewOp = value.getDefiningOp<ttg::MemDescSubviewOp>();
+               assert(subviewOp && "Failed to find TMEMAllocOp");
```

Comment:
The definiing op is not necessarily a MemDescSubviewOp. E.g, it could be an if-block. Can we leave the token field null when creating the TCGen5MMAOp op?